src/components/responsive-flyout/ResponsiveFlyout.js

```diff
@@ -0,0 +1,70 @@
+// @flow
+import * as React from 'react';
+import classNames from 'classnames';
+import { Flyout, Overlay } from '../flyout';
+import ScrollWrapper from '../scroll-wrapper';
+import type { FlyoutProps } from '../flyout/Flyout';
+import OverlayHeader from './OverlayHeader';
+
+import './styles/ResponsiveFlyout.scss';
+
+type Props = FlyoutProps & {
+    /** Components to render in the overlay */
+    children: ?React.Node,
+    /** Set className to the overlay wrapper */
+    className?: string,
+    /** Disable responsiveness behavior for higher-level components that want control */
+    enabled?: boolean,
+    /** Custom button to trigger for opening/closing the flyout */
+    flyoutButton: React.Element<any>,
+    /** Optional function to get the flyoutRef in parent components */
+    flyoutRefFn?: any => any,
+    /** Header for responsive layout */
+    header?: React.Element<any>,
+    /** Overlay classname */
+    overlayClassName?: string,
+    /** Optional function to get the scrollRef in parent components */
+    scrollRefFn?: any => any,
+};
+
+class ResponsiveFlyout extends React.Component<Props> {
+    static defaultProps = {
+        position: 'bottom-left',
+        enabled: true,
+    };
+
+    constructor(props) {
+        super(props);
+        this.flyoutRef = React.createRef();
+    }
+
+    render() {
+        const {
+            header,
+            flyoutButton,
+            children,
+            enabled,
+            scrollRefFn,
+            className,
+            overlayClassName,
+            flyoutRefFn,
+            ...rest
+        } = this.props;
+
+        return (
+            <Flyout ref={this.flyoutRef} className={classNames({ 'responsive-flyout': enabled }, className)} {...rest}>
+                {flyoutButton}
+                <Overlay className={overlayClassName}>
+                    <OverlayHeader onCloseClick={() => this.flyoutRef.current.closeOverlay()}>{header}</OverlayHeader>
+                    {children != null && (
+                        <ScrollWrapper scrollRefFn={scrollRefFn} shadowSize="contain">
```

Comment:
I don't think we want to assume the scroll behavior here. We currently allow consuming components to decide what goes in their Overlay.